Prepare for 1.38.0 release (#6441)
open-telemetry/opentelemetry-java: CHANGELOG.md

## Unreleased

### API

* Stabilize synchronous gauge
  ([#6419](https://github.com/open-telemetry/opentelemetry-java/pull/6419))

#### Incubator

* Add put(AttributeKey<T>, T) overload to EventBuilder
  ([#6331](https://github.com/open-telemetry/opentelemetry-java/pull/6331))

#### Baggage

* Baggage filters space-only keys
  ([#6431](https://github.com/open-telemetry/opentelemetry-java/pull/6431))

### SDK

* Add experimental scope config to enable / disable scopes (i.e. meter, logger, tracer)
  ([#6375](https://github.com/open-telemetry/opentelemetry-java/pull/6375))

#### Traces

* Add ReadableSpan#getAttributes
  ([#6382](https://github.com/open-telemetry/opentelemetry-java/pull/6382))
* Use standard ArrayList size rather than max number of links for initial span links allocation
  ([#6252](https://github.com/open-telemetry/opentelemetry-java/pull/6252))

#### Metrics

* Use low precision Clock#now when computing timestamp for exemplars
  ([#6417](https://github.com/open-telemetry/opentelemetry-java/pull/6417))

> * Update invalid instrument name log message to indicate that forward slash `/`

Reviewer: ```suggestion
* Update invalid instrument name log message now that forward slash `/` is valid
```
